bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/internal/core/Activator.java

```diff
@@ -158,4 +160,17 @@ public void stop(BundleContext bc) throws Exception {
 
     }
 
+    protected static void validateConnections(Rule r) {
```

Comment:
I would expect such a method in an utility class. Maybe you can move the extra null-checks to the ConnectionValidator and spend an extra method there e.g.: validateConnections(ModuleTypeRegistry moduleTypeRegistry, Rule rule)